### PR TITLE
Add smoketest builds for openvpn3, openvpn3-linux and ovpn-dco

### PR DESCRIPTION
**Pull request by mattock**

### CODE REVIEW

**Comment by mattock** [comment]
The openvpn3-linux smoketest builds include both OpenSSL and MbedTLS builds. If those builds pass then the rest of the builds are launched.

The openvpn3 and ovpn-dco smoketests include just one build, followed by the normal builds if all goes well.

I've tested each of these individually, but independent testing would be good.

**Review by Nomoney2worryabout — APPROVED** [review]

**Review by uddr — APPROVED** [review]

File: buildbot-host/buildmaster/master.cfg [hunk 1]
```diff
@@ -604,6 +607,22 @@ factories.update(
 )
 del factory
 
+# openvpn3 smoketest
+factory = util.BuildFactory()
+factory = openvpn3AddCommonLinuxStepsToBuildFactory(factory, combo)
```
Comment (flichtenheld) [inline buildbot-host/buildmaster/master.cfg]: What is `combo` here supposed to mean. We are not in a `for` loop, are we? I know you only copied this, but still would like to clean this up.

File: buildbot-host/buildmaster/master.cfg [hunk 2]
```diff
@@ -604,6 +607,22 @@ factories.update(
...
+            "types": ["openvpn3-smoketest"],
+            "schedulers": ["openvpn3-smoketest"],
+        }
+    }
+)
+del factory
+
 # Basic openvpn3 builds
 factory = util.BuildFactory()
 factory = openvpn3AddCommonLinuxStepsToBuildFactory(factory, combo)
```
Comment (flichtenheld) [inline buildbot-host/buildmaster/master.cfg]: Copied from here

File: buildbot-host/buildmaster/master.cfg [hunk 3]
```diff
@@ -656,6 +711,22 @@ factories.update(
 )
 del factory
 
+# ovpn-dco smoketest builds
+factory = util.BuildFactory()
+factory = ovpnDcoAddCommonLinuxStepsToBuildFactory(factory, combo)
```
Comment (flichtenheld) [inline buildbot-host/buildmaster/master.cfg]: `combo` again

File: buildbot-host/buildmaster/master.cfg [hunk 4]
```diff
@@ -656,6 +711,22 @@ factories.update(
...
+            "types": ["ovpn-dco-smoketest"],
+            "schedulers": ["ovpn-dco-smoketest"],
+        }
+    }
+)
+del factory
+
 # Basic ovpn-dco builds
 factory = util.BuildFactory()
 factory = ovpnDcoAddCommonLinuxStepsToBuildFactory(factory, combo)
```
Comment (flichtenheld) [inline buildbot-host/buildmaster/master.cfg]: Copied from here

**Review by flichtenheld — CHANGES_REQUESTED** [review]

File: buildbot-host/buildmaster/master.cfg [hunk 5]
```diff
@@ -655,6 +674,24 @@ factories.update(
 )
 del factory
 
+# MbedTLS openvpn3-linux smoketest builds
```
Comment (dsommers) [inline buildbot-host/buildmaster/master.cfg]: I'm not sure this makes sense any more.  As of `v22_dev`, mbed TLS support has essentially not been accounted for.

**Review by dsommers — CHANGES_REQUESTED** [review]
I would not consider mbed TLS at the moment.

**Review by Nomoney2worryabout — APPROVED** [review]

File: buildbot-host/buildmaster/master.cfg (hunk 5 above)
Comment (mattock) [reply buildbot-host/buildmaster/master.cfg]: Are you saying we should disable mbed TLS build altogether or?

File: buildbot-host/buildmaster/master.cfg (hunk 1 above)
Comment (mattock) [reply buildbot-host/buildmaster/master.cfg]: You're correct - copy and paste error. Not really sure what the implications are to be honest, but I'll clean it and the other places up.

File: buildbot-host/buildmaster/master.cfg (hunk 5 above)
Comment (flichtenheld) [reply buildbot-host/buildmaster/master.cfg]: For openvpn3-linux and openvpn3 there is no one maintaining the mbedTLS port. So if there are any problems with it we just disable the offending build, unless it is trivial to fix. In the case of openvpn3-linux I think there are no other known mbedTLS builds left. For openvpn3 we still have some internally.

Comment (flichtenheld) [reply buildbot-host/buildmaster/master.cfg]: So yes, I think disabling mbedTLS builds for openvpn3-linux makes sense until someone actually offers to maintain that "port".

Comment (dsommers) [reply buildbot-host/buildmaster/master.cfg]: Yes, I'm pretty sure of dropping mbed TLS support in OpenVPN 3 Linux. It's not even configurable at the moment with the Meson setup in use now.

I'm going to remove the mention of mbed TLS in the BUILD.md as well for the `v23` release.

**Comment by mattock** [comment]
I pushed three more commits that:

* Remove openvpn3-linux mbedtls builds (smoketest and normal)
* Remove unnecessary "combo" and "config_opts" parameters from openvpn3-linux, openvpn3 and ovpn-dco

I ran them through my private buildbot with one smoketest and one normal builder and all looks good afaics.

**Review by flichtenheld — APPROVED** [review]

**Pull request merged**